TrueDialog.Client.Standard/Retry/RetryPolicy.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Threading;
+
+namespace TrueDialog.Retry
+{
+    /// <summary>
+    /// Base class for all retry policies.
+    /// </summary>
+    class RetryPolicy<TDetector> : IRetryPolicy
+        where TDetector : IRetryDetectionStrategy, new()
+    {
+        private readonly RetryStrategy m_retryStrategy;
+
+        public RetryPolicy(RetryStrategy strategy)
+        {
+            m_retryStrategy = strategy;
+        }
+
+        private void InnerExecute(Action<int> action)
```

Comment:
We will need to rework this to use Task.Delay instead of Thread.Sleep.  Think we already have this done in our private libraries, should be okay to port that over to here.